src/test/java/org/sagebionetworks/bridge/participantroster/DownloadParticipantRosterWorkerProcessorTest.java

```diff
@@ -0,0 +1,169 @@
+        doReturn(new ArrayList<AccountSummary>()).when(mockBridgeHelper).getAccountSummariesForApp(
+                anyString(), anyString(), anyInt(), anyInt(), anyString());
+
+        processor.accept(makeValidRequestNode());
+
+        // verify that this call isn't reached, because study participant does not have a verified email
+        verify(mockBridgeHelper, never()).getAccountSummariesForApp(anyString(), anyString(), anyInt(), anyInt(), anyString());
+    }
+
+    @Test
+    public void getCsvHeaders(){
+        AccountSummary accountSummary = makeAccountSummary();
+        String[] headers = processor.getCsvHeaders(accountSummary.getAttributes());
+
+        assertEquals(headers.length, 3);
+        assertEquals("firstName", headers[0]);
+        assertEquals("lastName", headers[1]);
+        assertEquals("email", headers[2]);
+    }
+
+    @Test
+    public void getAccountSummaryArray() {
+        AccountSummary accountSummary = new AccountSummary();
+        accountSummary.putAttributesItem("firstName", "first-name");
+
+        String[] headers = processor.getCsvHeaders(accountSummary);
+        assertEquals(headers.length, 1);
+        assertEquals("attributes", headers[0]);
+
+        String[] accountSummaryArray = processor.getAccountSummaryArray(accountSummary, headers);
+        assertEquals(accountSummaryArray.length, 1);
+        assertEquals("{\"firstName\":\"first-name\"}", accountSummaryArray[0]);
+    }
+
+    @Test
+    public void normalCase() throws Exception {
+        StudyParticipant participant = new StudyParticipant();
+        participant.setEmail("example@example.org");
+        participant.setEmailVerified(true);
+        participant.addRolesItem(Role.RESEARCHER);
+
+        doReturn(participant).when(mockBridgeHelper).getParticipant(APP_ID, USER_ID, false);
+
+        when(mockBridgeHelper.getAccountSummariesForApp(anyString(), anyString(), anyInt(), anyInt(), anyString()))
+                .thenReturn(ImmutableList.of(makeAccountSummary(), makeAccountSummary(), makeAccountSummary()))
+                .thenReturn(new ArrayList<>());
+
+        doNothing().when(mockZipHelper).zipWithPassword(anyList(), any(File.class), anyString());
+
+        processor.accept(makeValidRequestNode());
+    }
+
+    private static ObjectNode makeValidRequestNode() {
+        ObjectNode requestNode = JSON_MAPPER.createObjectNode();
+        requestNode.put("appId", APP_ID);
+        requestNode.put("userId", USER_ID);
+        requestNode.put("password", PASSWORD);
+
+        return requestNode;
+    }
+
+    private static AccountSummary makeAccountSummary() {
+        AccountSummary accountSummary = new AccountSummary();
+        accountSummary.putAttributesItem("firstName", "first-name");
+        accountSummary.putAttributesItem("lastName", "last-name");
+        accountSummary.putAttributesItem("email", "email@email.com");
+
+        return accountSummary;
+    }
```

Comment:
Consider taking an int or a string suffix as an input so that we can test this with rows that have different information.

Consider adding null values so that you can test that's handled correctly.